src/simulationExecutor.ts

```diff
@@ -205,7 +205,7 @@ export class SimulationExecutor {
     } catch (error) {
       // re-simulate the transaction with `save` set to true so the failure can be analyzed later
       try {
-        simulate(true);
+        await simulate(true);
```

Comment:
When I was running the tests I've noticed that we might have missed waiting for this second simulation